doc/man3/RSA_get0_key.pod

```diff
@@ -43,6 +44,12 @@ RSA_set0_multi_prime_params, RSA_get_version
                                      const BIGNUM *coeffs[]);
  int RSA_set0_multi_prime_params(RSA *r, BIGNUM *primes[], BIGNUM *exps[],
                                 BIGNUM *coeffs[], int pnum);
+ int RSA_get0_all_params(RSA *r, STACK_OF(BIGNUM_const) *primes,
+                         STACK_OF(BIGNUM_const) *exps,
+                         STACK_OF(BIGNUM_const) *coeffs);
+ int RSA_set0_all_params(RSA *r, const STACK_OF(BIGNUM) *primes,
+                         const STACK_OF(BIGNUM) *exps,
+                         const STACK_OF(BIGNUM) *coeffs);
```

Comment:
Do you insist on internalizing them?

Reply:
Well, I may insist but I'm willing to hear arguments as to why we should make them public.

Reply:
Quite frankly, I find the other ones clumsy.  With those, you have to make two or three calls depending on the amount of primes:

``` C
RSA_set0_factors(rsa, p, q);
RSA_set0_crt_params(rsa, dP, dQ, qInv);
/* The following for multi-primes only */
RSA_set0_multi_prime_params(rsa, extra_primes, extra_exps, extra_coeffs);
```

when these new functions allow for just one call regardless of the number of primes et al:

``` C
RSA_set0_all_params(rsa, all_factors, all_exps, all_coeffs);
```

This is more like what an RSA key *is* outside of the ASN.1 constraints (the current calls are *very much* an effect of what needed to happen to the ASN.1 spec to make it backward compatible).

So most of all, my argument is convenience for our users.

I can see one thing that makes it less convenient, and it's the use of stacks.  I'm perfectly willing to make an array variant too (side note: it should be possible to extract an array from a stack, that would simplify so many things)

Reply:
While I understand where you are coming from we should perhaps have considered doing it this way before we introduced the RSA_set0_multi_prime_params() functions et al. We should be seeking stability in our API as much as possible and also seeking to restrict its growth. Every time we add new functions that's another set of functions we have to support for a very long time. Sometimes we *have*  to introduce new functions because there is no other way. But that isn't the case with this. I think we should not add this to the public API at this stage. We can always reconsider later if there is sufficient demand and reason for it.